### ISSUE TITLE
The current version of the app that is available on Google Play doesn't allow private webdav login

### ISSUE THREAD

**Issue by hiromipaw** [OpenArchive/Save-app-android#394]
The current version of the app available on google play gets stuck on "Logging in..." when configuring a private webdav space.

**To Reproduce**

1. Download the app from google play
1. Setup a private webdav space
2. Enter URL, username and password
3. The app is stuck on "Logging in..."

**Expected behavior**
A private webdav space should be setup allowing the user to upload images.

Will try to gather more information about android environment affected. This was tested on two different phones at the moment, one of this was a Galaxy s10.

**Comment by hiromipaw** [comment]
@purvi-ranawat can you reproduce this issue? @pkhivesara can you check?

**Comment by pkhivesara** [comment]
@hiromipaw What is the url for the webdav space that you are trying to login with?

**Comment by foundscapes** [comment]
https://nx27277.your-storageshare.de/remote.php/webdav @pkhivesara

**Comment by purvi-ranawat** [comment]
@foundscapes / @hiromipaw : After installing the Save app from Play Store, when I tried entering the URL "https://nx27277.your-storageshare.de/" and entered  the valid username and password, it is logging us in correctly. After clicking on the tick mark, the URL is appended with "https://nx27277.your-storageshare.de/remote.php/dav/"  automatically. But when I enter the URL mentioned above, it is just giving us the text Logging in and nothing else is displayed.

**Comment by hiromipaw** [comment]
@pkhivesara I have tried on iOS and I used the full url: https://nx27277.your-storageshare.de/remote.php/webdav

Nevertheless if the url is invalid I think the app needs to return some error to the user, or a timeout, and give them the possibility to re-enter the correct url.

Is that something that is handled in some library in android?

At the UX level I think we should think that either:
- we reformat the url correctly
or:
- we tell the user to use only the domain/subdomain without the path

**Comment by foundscapes** [comment]
Following up on #415 - @ceylana re-tested with the shortened https://nx27277.your-storageshare.de URL using correct credentials - gave a "login failed" error. The only difference from her test with the full URL was that no error message was displayed with the full URL and it just hung, whereas we got an error message this time.

**Comment by foundscapes** [comment]
I duplicated the same issue Alex is having on my android
OS: Android 12
Device: pixel 3a
App Version 0.2.5-BETA-4

**Comment by hiromipaw** [comment]
We have just made a new release and it's still an issue. 
App Version: 20525 (0.2.5-BETA-5)
@pkhivesara can you check?
@purvi-ranawat can you test the latest version?

**Comment by foundscapes** [comment]
> @foundscapes / @hiromipaw : After installing the Save app from Play Store, when I tried entering the URL "https://nx27277.your-storageshare.de/" and entered the valid username and password, it is logging us in correctly. After clicking on the tick mark, the URL is appended with "https://nx27277.your-storageshare.de/remote.php/dav/" automatically. But when I enter the URL mentioned above, it is just giving us the text Logging in and nothing else is displayed.

@purvi-ranawat can you retest on multiple android devices to see if you're having the same issue with the shortened url? It's odd both Alex and I would and you wouldn't. thank you!

**Comment by foundscapes** [comment]
@purvi-ranawat @hiromipaw Alex and I were both able to login with the shortened link after the recent update! Save 0.2.5-BETA-5 seems to have fixed this problem! Long link still doesn't work - it just hangs so we at least need an indicator message to tell people to shorten it if we can't make it possible for both links to work with login. ty all!

**Comment by foundscapes** [comment]
> @purvi-ranawat @hiromipaw Alex and I were both able to login with the shortened link after the recent update! Save 0.2.5-BETA-5 seems to have fixed this problem! Long link still doesn't work - it just hangs so we at least need an indicator message to tell people to shorten it if we can't make it possible for both links to work with login. ty all!

Update - I can no longer login with the short link and still cannot with the long link but Alex is able to login with either on her phone so I'm not sure why my google pixel 3a isn't doing it now with the version that launched yesterday 0.2.5-BETA-5

**Comment by purvi-ranawat** [comment]
@foundscapes / @hiromipaw : I am able to login with the short link " https://nx27277.your-storageshare.de/" but with the long link " https://nx27277.your-storageshare.de/remote.php/dav/", it still shows me Logging in only after entering the valid username and password.  Instead some valid error message should be displayed if user is not able to login. Also, as mentioned earlier, when I try the other long link "https://nx27277.your-storageshare.de/remote.php/webdav/", I am able to login.

So, the final results are : 

Able to login successfully with the below two URLS : 

1.  https://nx27277.your-storageshare.de/
2. https://nx27277.your-storageshare.de/remote.php/webdav/

Getting Logging In message for the below URL  : 
1. https://nx27277.your-storageshare.de/remote.php/webdav/

I have tried with two devices and getting the same results. Both are samsung phones : Samsung A13 and Samsung M32. I will try to arrange for more android devices and see if it is same

**Comment by foundscapes** [comment]
> 

I am still having the same problem with same credentials.

**Comment by foundscapes** [comment]
I am still having same issue and reproduced this on my phone as well today

OS version: Android 12
Device: Google Pixel 3a
App Version : Save 0.2.5-BETA-5

**Comment by foundscapes** [comment]
@purvi-ranawat we're now wondering if the issue is on older OS - can you test on a Google Pixel 3a with Android 12 OS  and the same version of Save 0.2.5-BETA-5? to make sure?

**Comment by purvi-ranawat** [comment]
@foundscapes : I have tried the below URLs on Samsung A70 - Android 11 and I am able to logging successfully . 

https://nx27277.your-storageshare.de/
https://nx27277.your-storageshare.de/remote.php/webdav/

Do you specifically need Google Pixel 3a phone with Android 12 OS?. If yes, I will buy the same

**Comment by foundscapes** [comment]
@purvi-ranawat Ok so we know it's still working on an even older version, but I would be curious to know if anyone can reproduce it in the same/similar environment. Please do not buy a phone, is there a way to do a simulation?

**Comment by purvi-ranawat** [comment]
@foundscapes : I am trying on Google Pixel 3a emulator if I am able to install Save on it. If I am successful, I will trying the issue on the same. I will let you know

**Comment by purvi-ranawat** [comment]
@foundscapes : I tried logging on Google Pixel 3a emulator which had Android 11 on it. I was abl to login successfully with the following URLs : 

https://nx27277.your-storageshare.de/
https://nx27277.your-storageshare.de/remote.php/webdav/

**Comment by foundscapes** [comment]
@purvi-ranawat ok can you test it on the simulator with OS Android 12? thanks!

**Comment by tladesignz** [comment]
Currently working on this. Get me a few more hours, then I'll release another build which should make this way more reliable and clear.

**Comment by tladesignz** [comment]
@purvi-ranawat: New release is out! (Might take a while to update.)

**Comment by purvi-ranawat** [comment]
@foundscapes : This is working fine on Google Pixel 3a - Android 12 emulator

Below is the link to the video : 

https://screencast-o-matic.com/watch/c0eDQpV44G6

**Comment by foundscapes** [comment]
@tladesignz @purvi-ranawat  - are you using the same login as I am (OA_Admin)?
Great thanks both! I just reinstalled it on the Google Pixel 3a OS 12 and I got the error "Unable to resolve host..." when I tried both long and short nxc27277...links to hetzner storage. 

I could not read the rest of the error popup bc it disappeared too quickly and didn't respond/expand when I tapped on it, please see photo (I can no longer take screenshots of Save on the phone now btw so got creative).
1) can you make it easier to read the error message
2) any idea what's causing it?  I will do a factory reset (again) to see if this helps but I cannot understand why I am the only one having issues. 

![SaveError_03_16_23](https://user-images.githubusercontent.com/2231711/225746186-b40ed43c-cc8f-46eb-b31c-e5ecdc9642dc.jpg)

@purvi-ranawat  - are you using the same login as I am (OA_Admin)?

**Comment by tladesignz** [comment]
@foundscapes, there's a typo in the hostname. It's "nx27277" not "ncx27277".

**Comment by foundscapes** [comment]
Aha, @tladesignz that was it - it's now working! This may have been the issue for the last few weeks but did not have any error before. However, I am glad it happened as the error message needs to expand as I couldn't read the rest of it, can we work on that? thank you!

**Comment by hiromipaw** [comment]
@foundscapes will be opening a new issue for the error message.
